app/src/main/java/com/example/henzoshimada/feeltrip/Mood.java

```diff
@@ -86,51 +96,60 @@ public Mood(String user){
         stateVector = new boolean[size];
         delState = false;
         made = null;
+        emoji = 0;
     }
 
     public String getId() {
-        if (id == null){ //mood does not yet exist within the Elasticsearch database
+        if (id == null) { //mood does not yet exist within the Elasticsearch database
             return "-1";
-        }
-        else {//return ID that the mood is stored under in the Elasticsearch database
+        } else {//return ID that the mood is stored under in the Elasticsearch database
             return id;
         }
     }
 
     public void setId(String id) {
-        this.id = id;
+        if (this.id != id) {
+            this.id = id;
+        }
     }
 
     // Evaluate stateVector and check if this mood event is changed.
-    public boolean isChanged(){
-        for (boolean value: stateVector){
-            if(value){return true;}
+    public boolean isChanged() {
+        for (boolean value : stateVector) {
+            if (value) {
+                return true;
+            }
         }
         return false;
     }
 
-    public boolean[] getAllState(){
+    public boolean[] getAllState() {
         return stateVector;
     }
 
-    public void resetState(){
+    public void resetState() {
         Arrays.fill(stateVector, false);
     }
 
     public boolean getStateByIndex(int index) {
         return stateVector[index];
     }
 
-    public void setStateByIndex(int index){
-        stateVector[index] = true;
+    public void setStateByIndex(int index) {
+        if (stateVector[index] != true) {
+            stateVector[index] = true;
+        }
     }
 
 
-    public String getUsername(){
+    public String getUsername() {
         return username;
     }
 
     public String getImage() {
+        if (image != null) {
+            return Html.fromHtml(image).toString();
```

Comment:
Return the HTML decoded characters as a single String. This allows us to edit the raw value of the image String without ruining its integrity